flexmeasures/cli/data_add.py

```diff
@@ -1117,6 +1125,293 @@ def add_schedule_for_storage(
+    if last_x_flag_given:
+        start, end = get_timerange_from_flag(
+            last_hour=last_hour,
+            last_day=last_day,
+            last_7_days=last_7_days,
+            last_month=last_month,
+            last_year=last_year,
+            timezone=timezone,
+        )
+
+    # apply offsets, if provided
+    if start_offset is not None:
+        if start is None:
+            start = now
+        start = apply_offset_chain(start, start_offset)
+
+    if end_offset is not None:
+        if end is None:
+            end = now
+        end = apply_offset_chain(now, end_offset)
+
+    # the case of not getting --start, --start-offset or any --last-X flag
+    if start is None:
+        click.secho(
+            "Either --start, --start-offset or any of the --last-X flags should be provided."
+            " Trying to use the latest datapoint of the report sensor as the start time and "
+            "the current time as the end...",
+            **MsgStyle.WARN,
+        )
+        last_value_datetime = (
+            db.session.query(func.max(TimedBelief.event_start))
+            .filter(TimedBelief.sensor_id == sensor.id)
+            .one_or_none()
+        )
+
+        # If there's data saved to the reporter sensors
+        if last_value_datetime[0] is not None:
+            start = last_value_datetime[0]
+        else:
+            click.secho(
+                f"Could not find any data for the report sensor {sensor}.",
+                **MsgStyle.ERROR,
+            )
+            raise click.Abort()
+
+    # the case of not getting --start, --start-offset or any --last-X flag
+    if end is None:
+        click.secho(
+            "Either --end, --end-offset or any of the --last-X flags should be provided."
+            " Trying to use the latest datapoint of the report sensor as the start time and "
+            "the current time as the end...",
+            **MsgStyle.WARN,
+        )
+        end = now
+
+    click.echo(f"Report scope:\n\tstart: {start}\n\tend: {end}")
+
+    click.echo(
+        f"Looking for the Reporter {reporter_class} among all the registered reporters...",
+    )
+
+    # get reporter class
+    ReporterClass = app.reporters.get(reporter_class)
+
+    # check if it exists
+    if ReporterClass is None:
+        click.secho(
+            f"Reporter class `{reporter_class}` not available.",
+            **MsgStyle.ERROR,
+        )
+        raise click.Abort()
+
+    click.secho(f"Reporter {reporter_class} found.", **MsgStyle.SUCCESS)
+
+    reporter_config_raw = json.load(reporter_config)
+
+    # initialize reporter class with the reporter sensor and reporter config
+    reporter: Type[Reporter] = ReporterClass(
+        sensor=sensor, reporter_config_raw=reporter_config_raw
+    )
+
+    click.echo("Report computation is running...")
+
+    # compute the report
+    result: Type[BeliefsDataFrame] = reporter.compute(
+        start=start, end=end, input_resolution=resolution
+    )
```

Comment:
Running this function for a period with no data gives an error. Currently, I get:
```
IndexError: list index out of range
```

From:
```
data/models/reporting/__init__.py", line 128, in compute
    result.index = result.index.set_levels(
```

I'd like an empty result instead, so the `Report computation done, but the report is empty.` warning can kick in.